### PR TITLE
doc: add the ARC part in release 2.3

### PR DESCRIPTION
**Pull request by vonhust**
add the ARC processor related part for release 2.3

Signed-off-by: Wayne Ren <wei.ren@synopsys.com>

### CODE REVIEW

**Review by ioannisg — APPROVED** [review]
Content-wise looks good to me, thanks.
Regarding MPU Gap Filling, you might want to write this feature enhancement in a more generic way, see, e.g. how it was done for ARM:

```
 Enhanced runtime MPU programming in ARMv8-M architecture, making
    the full partitioning of kernel SRAM memory a user-configurable
    feature.
```

File: doc/releases/release-notes-2.3.rst [hunk 1]
```diff
@@ -160,7 +160,14 @@ Architectures
 
 * ARC:
 
-  * <TBD>
+  * Changed to privilege stack generation for ARC MPU version 2 to avoid
```
Comment (ioannisg) [inline doc/releases/release-notes-2.3.rst:163]: This entry might be not 100% clear, might consider slight rewording.

File: doc/releases/release-notes-2.3.rst [hunk 2]
```diff
@@ -160,7 +160,14 @@ Architectures
...
+    memory waste
+  * Added MPU GAP FILLING for ARC MPU version 3
```
Comment (ruuddw) [inline doc/releases/release-notes-2.3.rst:165]: Suggestion: (thanks Ioannis!)
"Enhanced runtime programming for the MPU v3 by making the gap-filling of kernel memory a user-configurable
    feature"

**Review by carlescufi — APPROVED** [review]
Looks great to me. Thanks @vonhust

**Review by ruuddw — APPROVED** [review]

File: doc/releases/release-notes-2.3.rst [hunk 3]
```diff
@@ -160,7 +160,15 @@ Architectures
 
 * ARC:
 
-  * <TBD>
+  * Changed to automatic generation of privilege stack for ARC MPU V2 to
+    avoid the potential waste of memory When USERSPACE is configured
```
Comment (ioannisg) [inline doc/releases/release-notes-2.3.rst:164]: When -> when

**Review by abrodkin — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**